### PR TITLE
Filter sort

### PR DESCRIPTION
**Pull request by NtshVrm**
Added the following functionality:

**SORT BY**
- Most Popular(by default)
- Price Low to High
- Price High to Low
- Newest 

**FILTERS**
- By price range
- By Rating
- By Gender
- By Category

**PREVIEW**

chrome-capture (1)

### CODE REVIEW

File: src/assets/index.js [hunk 1]
```diff
@@ -23,6 +23,23 @@ import shirt from "./shirt.png";
...
+import cap_mercedes from "./cap-mercedes.png";
+import cap_redbull from "./cap-redbull.png";
+import ferrari_shirt from "./ferrari-shirt.png";
+import mclaren_womens from "./mclaren-womens.png";
+import merc_shirt from "./merc-shirt.png";
+import merc_womens from "./merc-womens.png";
+import redbull_shirt from "./redbull-shirt.png";
+import ferari_bag from "./ferrari-bag.png";
+import merc_bag from "./mercedes-bag.jpg";
+import redbull_bag from "./redbull-bag.jpg";
```
Comment (HarshithaKonduru214) [inline src/assets/index.js:42]: Can directly export the necessary images and use it.

File: src/components/productListing/ProductListing.css [hunk 2]
```diff
@@ -42,23 +42,27 @@
...
 }
 
 .sort {
-  border: none;
-  background-color: transparent;
-  font-size: 01rem;
+  border: 3px solid var(--primary-color);
+  border-radius: 2rem;
+  padding: 0.3rem;
+  background-color: white;
```
Comment (HarshithaKonduru214) [inline src/components/productListing/ProductListing.css]: Put it in variable.

File: src/components/productListing/ProductListing.css [hunk 3]
```diff
@@ -42,23 +42,27 @@
...
+  font-size: 1.5rem;
 }
 
 .listing-container {
   background-color: white;
```
Comment (HarshithaKonduru214) [inline src/components/productListing/ProductListing.css]: Same as above

File: src/components/productListing/FilterSidebar.jsx [hunk 4]
```diff
@@ -2,25 +2,43 @@ import { useState } from "react";
...
+            <input
+              type="range"
+              min="0"
+              max="300"
+              className="slider"
+              onChange={(event) => {
+                setPrice(event.target.value);
+                dispatch({ type: "RANGE_SORT", payload: event.target.value });
+              }}
+            />
```
Comment (znjs) [inline src/components/productListing/FilterSidebar.jsx:41]: can use `value` attribute to set a default position for range slider

File: src/components/productListing/FilterSidebar.jsx [hunk 5]
```diff
@@ -58,7 +76,17 @@ export function FilterSidebar() {
                 {item.options.map((obj) => {
                   return (
                     <div className="filter-option">
-                      <input type={item.type} name="sort" />
+                      <input
+                        type={item.type}
+                        name={item.title}
+                        onChange={(event) => {
+                          dispatch({
+                            type: "FILTER",
```
Comment (znjs) [inline src/components/productListing/FilterSidebar.jsx:84]: can use variables instead of strings to avoid typo errors

**Review by HarshithaKonduru214 — APPROVED** [review]

File: src/pages/ProductListing/ProductListing.jsx [hunk 6]
```diff
@@ -7,15 +7,18 @@ import {
...
 export default function ProductListing() {
+  const { finalData, dispatch } = useProduct();
   const [products, setProducts] = useState([]);
 
   useEffect(() => {
     async function getProducts() {
       try {
         const output = await axios.get("/api/products");
         setProducts(output.data.products);
+        dispatch({ type: "DATA", payload: output.data.products });
```
Comment (znjs) [inline src/pages/ProductListing/ProductListing.jsx:21]: can use variables instead of strings

File: src/reducer/product-reducer.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
+const productReducer = (state, action) => {
+  const initial = state;
+  switch (action.type) {
+    case "DATA": {
```
Comment (znjs) [inline src/reducer/product-reducer.js:4]: can use variables instead of strings in whole reducer file

File: src/utils/product-utils.js [hunk 8]
```diff
@@ -0,0 +1,77 @@
+function getSortedData(productsList, sortBy) {
+  switch (sortBy) {
+    case "priceHighToLow": {
```
Comment (znjs) [inline src/utils/product-utils.js:3]: can use variables instead of strings in utils file

**Review by znjs — APPROVED** [review]
Good folder structure
Clean UI
functionality working as expected

**Review by ayush65 — APPROVED** [review]
Good folder structure
UI looks clean

**Review by Nikhil-Nirmal — APPROVED** [review]

**Pull request merged**